### DISCUSSION TITLE
stateTime vs refetchOnMount to avoid refetch in each component mount

### DISCUSSION THREAD

**Discussion by crisfcodes** [TanStack/query#4533] [Q&A]
I have a request which I run when the component is mounted, what I want is to avoid this request running in each of the component re-mounts. I have reviewed the [documentation](https://tanstack.com/query/v4/docs/reference/useQuery) and I have found two possible solutions, `stateTime: Infinity` and `refetchOnMount: false`, with both options I get the component to only run the request once, I want to know the pros and cons of each option or if there is any option that I have not taken into account that fits better for this use case.

@TkDodo

**Comment by TkDodo** [comment]
https://tkdodo.eu/blog/react-query-as-a-state-manager
